src/lib.rs

```diff
@@ -239,6 +266,8 @@ pub mod pallet {
 		IllegalMove,
 		BetTooLow,
 		BetDoesNotExist,
+		MatchNotOnGoing,
+		MatchNotExpired,
 	}
 
 	#[pallet::hooks]
```

Comment:
we can remove the hooks, since this solution is meant to replace `on_initialize`